### PR TITLE
MDEV-11045 systemd reload / rpm {pre|post}{inst|uninst}

### PR DESCRIPTION
**Pull request by grooverdan**
I think 3b430c8 should be revert once it hits 10.3 unstable branch to make mariadb exhibit consistent packaging behaviour to other distro packages as per the systemd presets https://www.freedesktop.org/wiki/Software/systemd/Preset/ which is what `%systemd_post %{daemon_name}.service` expands to.

The original MDEV-11045 asked for a daemon-reload however given the date of creating I suspect this is addressed by the following systemd macros commit and has made it into RPM distros.
https://github.com/systemd/systemd/commit/873e413323dfff4023604849c70944674ae5cd29

This also clears the way for the fixing of MDEV-10797.

I submit this under the MCA.

### CODE REVIEW

**Comment by svoj** [comment]
Hi @grooverdan, thanks for your contribution!

File: support-files/mysql.server.sh [hunk 1]
```diff
@@ -365,6 +365,19 @@ case "$mode" in
...
+    # restart if already running
+    # concept ported from Fedora and used to restart in package upgrade
+    # where we want the restart only if the server is currently running
+    if test -f "$lock_file_path"
+    then
+      $0 restart
+    else
+      exit 0
+    fi
+    ;;
```
Comment (vuvova) [inline support-files/mysql.server.sh]: lovely

File: support-files/rpm/server-postin.sh [hunk 2]
```diff
@@ -1,3 +1,7 @@
+%if %{with init_systemd}
+%systemd_post %{daemon_name}.service
+%endif
+
 if [ -f /usr/lib/systemd/system/mariadb.service -a -x /usr/bin/systemctl ]; then
```
Comment (vuvova) [inline support-files/rpm/server-postin.sh]: Is this run-time systemd detection needed anymore?

File: support-files/rpm/server-postun.sh [hunk 3]
```diff
@@ -1,15 +1,14 @@
...
 if [ $1 -ge 1 ]; then
-  if [ -x %{_sysconfdir}/init.d/mysql ] ; then
+  if [ -x /sbin/servce ] ; then
+    /sbin/service %{daemon_name} condrestart >/dev/null 2>&1 || :
+  elif [ -x %{_sysconfdir}/init.d/%{daemon_name} ] ; then
     # only restart the server if it was alredy running
-    if %{_sysconfdir}/init.d/mysql status > /dev/null 2>&1; then
-      %{_sysconfdir}/init.d/mysql restart
+    if %{_sysconfdir}/init.d/%{daemon_name} status > /dev/null 2>&1; then
+      %{_sysconfdir}/init.d/%{daemon_name} restart
```
Comment (vuvova) [inline support-files/rpm/server-postun.sh]: Huh? You've added `condrestart` in this very PR, why you're not using it?

File: support-files/rpm/server-postin.sh [hunk 4]
```diff
@@ -1,3 +1,7 @@
+%if %{with init_systemd}
+%systemd_post %{daemon_name}.service
```
Comment (vuvova) [inline support-files/rpm/server-postin.sh]: Are these systemd macros guaranteed to exist even on the oldest rpm-based systemd distro that we support?

**Comment by grooverdan** [comment]
Closed MDEV

**Pull request closed without merging**